### ISSUE TITLE
"Error! Binary hash does NOT match." when cross-building arm64 executables on x64 hosts

### ISSUE THREAD

**Issue by Niek** [vercel/pkg#1402]
### What version of pkg are you using?

5.4.1

### What version of Node.js are you using?

17.1.0

### What operating system are you using?

Debian 11

### What CPU architecture are you using?

x64

### What Node versions, OSs and CPU architectures are you building for?

arm64

### Describe the Bug

I can't build arm64 executables on x64 hosts:

```bash
$ docker run --rm -it --platform=linux/amd64 node bash -c 'touch test.js && npx -y pkg -t node14-linux-arm64 test.js'
> pkg@5.4.1
> Fetching base Node.js binaries to PKG_CACHE_PATH
  fetched-v14.18.1-linux-arm64        [====================] 100%

  fetched-v14.18.1-linuxstatic-arm64  [====================] 100%

> Error! Binary hash does NOT match.
```

Same error with `node12`, `node16` gives another error:
```bash
> Error! 404: Not Found
> Not found in remote cache:
  {"tag":"v3.2","name":"node-v16.12.0-linuxstatic-arm64"}
> Building base binary from source:
  built-v16.12.0-linuxstatic-arm64
> Error! Not able to build for 'linuxstatic' here, only for 'linux'
```

### Expected Behavior

On arm64 hosts, the building works fine:
```bash
$ docker run --rm -it --platform=linux/arm64 node bash -c 'touch test.js && npx -y pkg -t node14-linux-arm64 test.js && ls -lha test'
> pkg@5.4.1
> Fetching base Node.js binaries to PKG_CACHE_PATH
  fetched-v14.18.1-linux-arm64        [====================] 100%

npm notice 
npm notice New patch version of npm available! 8.1.2 -> 8.1.3
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.3
npm notice Run npm install -g npm@8.1.3 to update!
npm notice 
-rwxr-xr-x 1 root root 35M Nov 15 16:34 test
```

### To Reproduce

See the reproducible docker run above.

**Comment by Niek** [comment]
Probably related to: https://github.com/vercel/pkg/issues/1358

**Comment by Niek** [comment]
Pinning `pkg` to `5.4.0` works as a workaround. Waiting for https://github.com/vercel/pkg-fetch/issues/223 to be merged so it will be fixed.

**Comment by jesec** [comment]
Fixed with 5.5.1.

**Comment by Niek** [comment]
I can confirm that this is fixed now for Node 12, 14 and 16, Thanks!
